src/glib/mine/signalproc.h

```diff
@@ -856,6 +855,109 @@ class TOnlineHistogram {
 };
 
 /////////////////////////////////////////////////
+///   TDigest
+///   Data structure useful for percentile and quantile estimation for online data streams.
+///   It can be added to any anomaly detector to set the number of alarms triggered as a percentage of the total samples.
+///   This is the Data Lib Sketch Implementation: https://github.com/vega/datalib-sketch/blob/master/src/t-digest.js
+///    Paper: Ted Dunning, Otmar Ertl - https://github.com/tdunning/t-digest/blob/master/docs/t-digest-paper/histo.pdf
+class TTDigest {
+private:
+	TInt Nc;
+	TInt Size;
+	TInt Last;
+	TFlt TotalSum;
+	TFltV Weight;
+	TFltV Mean;
+	TFlt Min;
+	TFlt Max;
+	// double buffer to simplify merge operations
+	// MergeWeight also used for transient storage of cumulative weights
+	TFltV MergeWeight;
+	TFltV MergeMean;
+	// temporary buffers for recently added values
+	TInt Tempsize;
+	TFlt UnmergedSum;
+	TInt TempLast;
+	TFltV TempWeight;
+	TFltV TempMean;
+	TFlt EPSILON;
```

Comment:
I don't think this guy is defined anywhere. Also, we don't need the DEFAULT_CENTROIDS - why not use default arguments in Init functions ?